torch/export/_trace.py

```diff
@@ -147,7 +147,6 @@ def _override_composite_implicit_decomp(ops_to_preserve, decomp_table, safe=True
 
     saved_tables = {}
     patched_ops = set()
-    removed_decomps = {}
```

Comment:
the refactor sounds reasonable to me (unifying the user API behind a single decomp table again, and documenting the "set decomp to None" behavior.

Although while you're at it, maybe we should clean up the internals and kill the separate `preserve_op` list that gets plumbed all the way to here (and instead, just have this function loop through the decomp table and check which ones are set to `None` to know which CompositeImplicits to override)

Reply:
Yep i wanted to see how CI goes with this change. I think there are two things we should do here:
1. If it is None, we check if the op is CIA and then try to preserve
2. If it is None, we check if the op is non CIA and then just delete from python decomp